src/PositionUtilities.php

```diff
@@ -26,26 +26,16 @@ public static function getRangeFromPosition($pos, $length, $text) {
      * @return LineCharacterPosition
      */
     public static function getLineCharacterPositionFromPosition($pos, $text) : LineCharacterPosition {
-        if ($pos >= \strlen($text)) {
-            $pos = \strlen($text);
+        $length = \strlen($text);
+        if ($pos >= $length) {
+            $pos = $length;
         } elseif ($pos < 0) {
             $pos = 0;
         }
 
-        $newlinePositions = [];
-        $newlinePos = -1;
-        while ($newlinePos = \strpos($text, "\n", $newlinePos + 1)) {
-            if ($newlinePos < $pos) {
-                $newlinePositions[] = $newlinePos;
-                continue;
-            }
-            break;
-        }
-
-        $lastNewline = \count($newlinePositions) - 1;
-        $char = $pos - ($lastNewline >= 0 ? $newlinePositions[$lastNewline] + 1 : 0);
-        $line = \count($newlinePositions);
-
+        $lastNewlinePos = \strrpos($text, "\n", -($length - $pos));
+        $char = $pos - ($lastNewlinePos === false ? 0 : $lastNewlinePos + 1);
+        $line = $pos > 0 ? \substr_count($text, "\n", 0, $pos) : 0;
```

Comment:
Wow - at first I thought the ternary was some micro-optimization to avoid the function call, but turns out that no, `substr_count` just has some non-intuitive behavior.

@nikic - According to the [docs](http://php.net/manual/en/function.substr-count.php), only a negative length should count from the end of haystack, but that's not what the function is [actually doing.]( https://github.com/php/php-src/blob/cbc416e8d34d7f653e9d31d450c01bb208aa7921/ext/standard/string.c#L5279-L5281) I just dug into it, and neither the [RFC](https://wiki.php.net/rfc/negative-string-offsets), nor the corresponding PR reference the expected behavior when zero-length is specified. Additionally, there aren't any tests for the zero-length case in `substr_count_basic.phpt`, so it strikes me as a simple oversight. What's PHP's stance on backwards compatibility for things like this? Docs bug or php bug?